python_modules/dagster/dagster_tests/definitions_tests/test_i_definition_prop_class_matches_init_adherence.py

```diff
@@ -0,0 +1,37 @@
+from inspect import signature
+from typing import List, Type
+
+from dagster._core.definitions import asset_checks
+from dagster._core.definitions.asset_checks import (
+    AssetChecksDefinition,
+    AssetChecksDefinitionProps,
+    IDefinitionPropsClassMatchesInit,
+)
+from dagster._utils.test import get_all_direct_subclasses_of_marker
+
+
+def test_adherence_of_all_idef_props_class_matches_init() -> None:
+    def_types: List[Type] = get_all_direct_subclasses_of_marker(
+        IDefinitionPropsClassMatchesInit, module=asset_checks
+    )
+
+    hard_coded_list = [
+        (AssetChecksDefinition, AssetChecksDefinitionProps),
+    ]
+
+    # keep hard_coded_list up-to-date with all subclasses of IDefinitionPropsClassMatchesInit
+    assert set([def_type.__name__ for def_type in def_types]) == set(
+        [entry[0].__name__ for entry in hard_coded_list]
+    ), (
+        "You likely added a subclass of IDefinitionPropsClassMatchesInit without adding it to the"
+        " hard_coded_list"
+    )
```

Comment:
this is just here because you couldn't figure out how to pluck `TPropsType` &  `TDefType` from the class object?